### PR TITLE
Replace godoc.org URLs

### PR DESCRIPTION
**Pull request by JamieEdge**
[godoc.org](https://godoc.org) now redirects to [pkg.go.dev](https://pkg.go.dev) (see [Redirecting godoc.org requests to pkg.go.dev](https://blog.golang.org/godoc.org-redirect)).

This pull request replaces all godoc.org references with equivalent pkg.go.dev URLs.

### CODE REVIEW

File: site/content/dependencies/_index.en.md [hunk 1]
```diff
@@ -4,8 +4,8 @@ weight = 30
 +++
 
 Mage supports a makefile-style tree of dependencies using the helper library
-[github.com/magefile/mage/mg](https://godoc.org/github.com/magefile/mage/mg). To
-declare dependencies, pass any number of dependent functions to mg.Deps.
+[mg](https://pkg.go.dev/github.com/magefile/mage/mg). To declare dependencies,
```
Comment (arschles) [inline site/content/dependencies/_index.en.md:7]:
```suggestion
[github.com/magefile/mage/mg](https://pkg.go.dev/github.com/magefile/mage/mg). To declare dependencies,
```

Comment (natefinch) [reply site/content/dependencies/_index.en.md:7]: This looks good with the given change

**Comment by natefinch** [comment]
Hey sorry, can you merge main again when you get the chance, and/or check the box that lets me do it? :)

**Comment by JamieEdge** [comment]
Sure, updated now.

**Comment by ladydascalie** [comment]
@natefinch I believe this is ready to go now, just needs workflow approval

**Review by arschles — APPROVED** [review]

**Pull request merged**